great_expectations/datasource/new_datasource.py

```diff
@@ -204,6 +220,33 @@ def get_batch_list_from_batch_request(
                 batches.append(new_batch)
             return batches
 
+    def _update_missing_data_asset_name_from_configured_data_connector(
+        self, batch_request: Union[BatchRequest, RuntimeBatchRequest]
+    ):
```

Comment:
Nitpick - could we add `-> None`? and general type annotations to any new functions?